packages/shared/src/contexts/NotificationsContext.tsx

```diff
@@ -28,31 +32,57 @@ export const NotificationsContextProvider = ({
   children,
   unreadCount = 0,
 }: NotificationsContextProviderProps): ReactElement => {
+  const isExtension = !!process.env.TARGET_BROWSER;
   const { user } = useContext(AuthContext);
-  const [hasPermission, setHasPermission] = useState(
-    globalThis.window?.Notification?.permission === 'granted',
-  );
-
-  const notificationsAvailable = () => {
-    return !!globalThis.window?.Notification;
-  };
+  const [isInitialized, setIsInitialized] = useState(false);
+  const [hasPermission, setHasPermission] = useState(false);
 
-  const requestPermission = async (): Promise<NotificationPermission> => {
+  const onTogglePermission = async (): Promise<NotificationPermission> => {
     if (!user) return 'default';
 
+    if (hasPermission) {
+      await OneSignalReact.setSubscription(false);
+      setHasPermission(false);
+      return 'denied';
+    }
```

Comment:
What's the purpose of this snippet?

Reply:
This used to be an abstraction of the actual `requestPermission`, there are areas where it waits for a response on what actually happened but we can turn this though into a boolean to keep it simple.

Reply:
This is for unsubscribing. Let me turn this to a boolean value to return so it will be clearer for the consuming end.